From f484c927a9fe5a54d36b3e9e752eda850b1ff27e Mon Sep 17 00:00:00 2001
From: Naveen Manohar <naveen.m@intel.com>
Date: Tue, 3 Apr 2018 11:50:01 +0530
Subject: [PATCH 1327/1582] FROMLIST: ASoC: Intel: glk: Add DAI links for
 Multi-Playback

Add new FE DAI link to support: Parallel playback on 2 ports simultaneously

LINK=https://patchwork.kernel.org/patch/10322067/

BUG=b:75996084
TEST=build and boot. verify basic audio isn't regressed.

Change-Id: I61a3f354c8ebbbd558e58117cb41645dc6bccdca
Signed-off-by: Naveen Manohar <naveen.m@intel.com>
Reviewed-on: https://chromium-review.googlesource.com/995715
Commit-Ready: Naveen M <naveen.m@intel.com>
Tested-by: Naveen M <naveen.m@intel.com>
Reviewed-by: Sathyanarayana Nujella <sathyanarayana.nujella@intel.com>
Reviewed-by: Furquan Shaikh <furquan@chromium.org>
Reviewed-by: Justin TerAvest <teravest@chromium.org>
---
 sound/soc/intel/boards/bxt_da7219_max98357a.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/sound/soc/intel/boards/bxt_da7219_max98357a.c b/sound/soc/intel/boards/bxt_da7219_max98357a.c
index 9a2f7f3..17f4ad9 100644
--- a/sound/soc/intel/boards/bxt_da7219_max98357a.c
+++ b/sound/soc/intel/boards/bxt_da7219_max98357a.c
@@ -51,6 +51,7 @@ struct bxt_card_private {
 enum {
 	BXT_DPCM_AUDIO_PB = 0,
 	BXT_DPCM_AUDIO_CP,
+	BXT_DPCM_AUDIO_HS_PB,
 	BXT_DPCM_AUDIO_REF_CP,
 	BXT_DPCM_AUDIO_DMIC_CP,
 	BXT_DPCM_AUDIO_HDMI1_PB,
@@ -571,6 +572,17 @@ static struct snd_soc_dai_link geminilake_dais[] = {
 		.dpcm_capture = 1,
 		.ops = &broxton_da7219_fe_ops,
 	},
+	[BXT_DPCM_AUDIO_HS_PB] = {
+		.name = "Bxt Audio Headset Playback",
+		.stream_name = "Headset Playback",
+		.cpu_dai_name = "System Pin2",
+		.platform_name = "0000:00:0e.0",
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.dpcm_playback = 1,
+		.nonatomic = 1,
+		.dynamic = 1,
+	},
 	[BXT_DPCM_AUDIO_REF_CP] = {
 		.name = "Bxt Audio Reference cap",
 		.stream_name = "Refcap",
-- 
2.7.4

